CMakeLists.txt

```diff
@@ -756,6 +756,11 @@ else()
   # is fixed in the code (Issue #847), force compiler to be conservative.
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fno-strict-aliasing")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-strict-aliasing")
+  
+  # Warnings, that when ignored are so severe, that they can segfault or even UB any application.
+  set(WARNINGS_AS_ERRORS "-Werror=switch -Werror=return-type")
```

Comment:
From the [Gcc docs](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html):

> Warn whenever a switch statement has an index of enumerated type and lacks a case for one or more of the named codes of that enumeration. (The presence of a default label prevents this warning.) case labels outside the enumeration range also provoke warnings when this option is used (even if there is a default label). This warning is enabled by -Wall. 

The project should already have `-Wall` enabled, so this warning should already be enabled. And a `default` case disables the warning; the compiler doesn't track existing labels for new ones.

Reply:
Yes, `-Wall` is enabled. Once the `switch` warning is treated as error, I could then go through all the switch/cases and remove the `default:` labels, triggering a compilation warning treated as error, if a newly case was forgotten to be handled.

Maybe not all `default:` cases will be removable, but any default case, which throws an exception due to unhandled case is a target, as an example.